### ISSUE TITLE
pypeit_setup fails for FIRE data. ValueError: Duplicate column names.

### ISSUE THREAD

**Issue by elNava** [pypeit/PypeIt#1636]
Hi!

As suggested in the slack channel, this issue is to report what might be a bug in the `pypeit_setup` step when one try to read FIRE data. I got this error message when I was executing the setup on the data directory:

```
[INFO]    :: Using metadata to determine unique configurations.
[INFO]    :: Unique configurations ignore frames with type: ['bias', 'dark']
[INFO]    :: Found 1 unique configuration(s).
Traceback (most recent call last):
  File "/home/benjamin/anaconda3/envs/pypeit/bin/pypeit_setup", line 8, in <module>
    sys.exit(Setup.entry_point())
             ^^^^^^^^^^^^^^^^^^^
  File "/home/benjamin/anaconda3/envs/pypeit/lib/python3.11/site-packages/pypeit/scripts/scriptbase.py", line 112, in entry_point
    cls.main(cls.parse_args())
  File "/home/benjamin/anaconda3/envs/pypeit/lib/python3.11/site-packages/pypeit/scripts/setup.py", line 117, in main
    pypeit_files = ps.fitstbl.write_pypeit(output_path=output_path, cfg_lines=ps.user_cfg, 
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/benjamin/anaconda3/envs/pypeit/lib/python3.11/site-packages/pypeit/metadata.py", line 1673, in write_pypeit
    subtbl = self.table[output_cols][in_cfg]
             ~~~~~~~~~~^^^^^^^^^^^^^
  File "/home/benjamin/anaconda3/envs/pypeit/lib/python3.11/site-packages/astropy/table/table.py", line 2063, in __getitem__
    out = self.__class__(
          ^^^^^^^^^^^^^^^
  File "/home/benjamin/anaconda3/envs/pypeit/lib/python3.11/site-packages/astropy/table/table.py", line 870, in __init__
    init_func(data, names, dtype, n_cols, copy)
  File "/home/benjamin/anaconda3/envs/pypeit/lib/python3.11/site-packages/astropy/table/table.py", line 1239, in _init_from_list
    self._init_from_cols(cols)
  File "/home/benjamin/anaconda3/envs/pypeit/lib/python3.11/site-packages/astropy/table/table.py", line 1467, in _init_from_cols
    self._make_table_from_cols(self, newcols)
  File "/home/benjamin/anaconda3/envs/pypeit/lib/python3.11/site-packages/astropy/table/table.py", line 1532, in _make_table_from_cols
    raise ValueError("Duplicate column names")
ValueError: Duplicate column names
```

Disscussing with Emma Softich in the slack channel, she found that there was a duplicated `calib` column in the scripts processing the input for the `.pypeit` file. More specifically, if you print the `output_cols` variable in the `metadata.py` script (line 1672), you will find:

```
[INFO]    :: Unique configurations ignore frames with type: ['bias', 'dark']
[INFO]    :: Found 1 unique configuration(s).
['filename', 'frametype', 'ra', 'dec', 'target', 'dispname', 'decker', 'binning', 'mjd', 'airmass', 'exptime', 'dichroic', 'calib', 'comb_id', 'bkg_id', 'calib']
Traceback (most recent call last):
  File "/home/benjamin/anaconda3/envs/pypeit/bin/pypeit_setup", line 8, in <module>
    sys.exit(Setup.entry_point())
             ^^^^^^^^^^^^^^^^^^^
```

where you can note the duplicated 'calib' key. As far as I have tested, this happens only with FIRE Echelle data. I am using the 1.13.0 version of PypeIt, and the issue is not present in previous versions. I tried dropping one 'calib' key manually and the code works, but it is not a proper solution since I would have to change the code everytime I want to use data from another spectrograph. Also, I tested if there is any difference between keeping the first 'calib' element or the second and it's the same.

I hope you can check and/or solve this issue for the next FIRE users to skip this problem.
Thanks!
Benjamín Navarrete

**Comment by debora-pe** [comment]
This issue is resolved in `1.13.1dev`.
This is how it looks running `pypeit_setup` on `1.13.1dev`:
```
pypeit_setup -r FIRE/ -s magellan_fire -c all
[INFO]    :: Adding metadata for fire_0029.fits.gz
[INFO]    :: Adding metadata for fire_0030.fits.gz
[INFO]    :: Adding metadata for fire_0031.fits.gz
[INFO]    :: Adding metadata for fire_0032.fits.gz
[INFO]    :: Adding metadata for fire_0033.fits.gz
[INFO]    :: Adding metadata for fire_0035.fits.gz
[INFO]    :: Adding metadata for fire_0036.fits.gz
[INFO]    :: Adding metadata for fire_0037.fits.gz
[INFO]    :: Adding metadata for fire_0038.fits.gz
[INFO]    :: Adding metadata for fire_0039.fits.gz
[INFO]    :: Adding metadata for fire_0040.fits.gz
[INFO]    :: Adding metadata for fire_0041.fits.gz
[INFO]    :: Adding metadata for fire_0042.fits.gz
[INFO]    :: Typing files
[WARNING] :: Cannot determine if frames are of type align.
[WARNING] :: Cannot determine if frames are of type dark.
[WARNING] :: Cannot determine if frames are of type illumflat.
[WARNING] :: Cannot determine if frames are of type lampoffflats.
[WARNING] :: Cannot determine if frames are of type sky.
[INFO]    :: Couldn't identify the following files:
[INFO]    :: fire_0037.fits.gz
[INFO]    :: Typing completed!
[INFO]    :: Using metadata to determine unique configurations.
[INFO]    :: Unique configurations ignore frames with type: ['bias', 'dark']
[INFO]    :: Found 1 unique configuration(s).
[INFO]    :: Unique configurations ignore frames with type: ['bias', 'dark']
[INFO]    :: PypeIt file successfully vetted.
[INFO]    :: PypeIt input file written to: /Users/dpelliccia/Desktop/test/magellan_fire_A/magellan_fire_A.pypeit
[INFO]    :: Calibration association file written to: /Users/dpelliccia/Desktop/test/magellan_fire_A/magellan_fire_A.calib
```

And I confirmed it failed on `1.13.0`

@elNava it would be good if you could jump on the `develop` branch and check it. Otherwise it would be available in the next release.

But I would close this issue, if it's OK with you.

**Comment by debora-pe** [comment]
This was fixed in `1.13.1dev` and will be available in the next release. Closing this issue.
